File: Moxion.Application/Shared/Simulators/Params/MotionSimulationParam.cs
using Moxion.Application.Abstractions.Simulators;
using Moxion.Common.Values;
using Moxion.Common.Values.Derived;

namespace Moxion.Application.Shared.Simulators.Params;

internal readonly record struct MotionSimulationParam(
  Position Displacement,
  Velocity Velocity,
  Acceleration Acceleration,
  Jerk Jerk,
  int DataCount
) : ISimulationParam;

File: Moxion.Presentation.Web/ClientApp/src/models/kinematics/MotionSimulationParam.ts
import Position from "../values/Position";
import Velocity from "../values/Velocity";
import Acceleration from "../values/Acceleration";
import Jerk from "../values/Jerk";
import TimeUnit from "../enumerations/TimeUnit";

class MotionSimulationParam {
    displacement: Position
    velocity: Velocity
    acceleration: Acceleration
    jerk: Jerk
    timeIntervalUnit: TimeUnit
    dataCount: number

    constructor(
        displacement?: Position,
        velocity?: Velocity,
        acceleration?: Acceleration,
        jerk?: Jerk,
        timeIntervalUnit?: TimeUnit,
        dataCount?: number
    ) {
        this.displacement = displacement ?? new Position()
        this.velocity = velocity ?? new Velocity()
        this.acceleration = acceleration ?? new Acceleration()
        this.jerk = jerk ?? new Jerk()
        this.timeIntervalUnit = timeIntervalUnit ?? 'Second'
        this.dataCount = dataCount ?? 0
    }
}

export default MotionSimulationParam